src/datepicker/datepicker.js

```diff
@@ -629,7 +629,16 @@ function(scope, element, attrs, $compile, $parse, $document, $rootScope, $positi
         scope.$parent.$watch(getAttribute, function(value) {
           scope.watchData[key] = value;
           if (key === 'minDate' || key === 'maxDate') {
-            cache[key] = new Date(value);
+            var date = parseDate(value);
+            if (!date) {
+              date = new Date(value);
+              if (angular.isDate(date) && !isNaN(date)) {
+                var dateArray = date.toISOString().split('T')[0].split('-');
```

Comment:
the reason I'm using the `toISOString()` (I plan to switch to equivalent `toJSON()` next commit) is to get the UTC version of the date, otherwise there is the same timezone issue we are seeing in #3437... the date for EST looks something like 9PM the day before, and the date becomes off by 1.